portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/IdentityStoreClientServiceImpl.java

```diff
@@ -341,24 +352,139 @@ public String getPrimaryDomainName() throws UserPortalUIException {
         return users;
     }
 
-//    @Override
-//    public String getPrimaryDomainName() throws UserPortalUIException {
-//        String primaryDomain;
-//        try {
-//            primaryDomain = getRealmService().getIdentityStore().getPrimaryDomainName();
-//        } catch (IdentityStoreException e) {
-//            String error = "Failed to get the primary domain name.";
-//            LOGGER.error(error, e);
-//            throw new UserPortalUIException(error);
-//        }
-//        return primaryDomain;
-//
-//    }
+    @Override
+    public List<UserListBean> getFilteredList(int offset, int length,
+                                              String claimURI, String claimValue, String domainName,
+                                              List<ClaimConfigEntry> requestedClaims) throws UserPortalUIException {
+
+        List<UserListBean> userList;
+        List<User> users;
+
+        if (length < 0) {
+            length = MAX_RECORD_LENGTH;
+        }
+
+        if (StringUtils.isNullOrEmpty(domainName)) {
+            domainName = getPrimaryDomainName();
+        }
+
+        if (StringUtils.isNullOrEmpty(claimURI)
+                || StringUtils.isNullOrEmpty(claimValue)) {
+            return getUserList(offset, length, domainName, requestedClaims);
+        } else {
+            MetaClaim metaClaim = new MetaClaim();
+            metaClaim.setClaimUri(claimURI);
+            try {
+                users = getRealmService().getIdentityStore().listUsers(
+                        metaClaim, claimValue, offset, length, domainName);
+            } catch (IdentityStoreException e) {
+                String error = "Error while retrieving users for " + claimURI + "= " + claimValue;
+                LOGGER.error(error, e);
+                throw new UserPortalUIException(error);
+            } catch (IllegalStateException e) {
+                String error = e.getMessage();
+                LOGGER.error(error, e);
+                throw new UserPortalUIException(error);
+            }
+        }
+
+        return generateUserListBean(users, requestedClaims);
+
+    }
+
+    @Override
+    public List<UserListBean> getUserList(int offset, int length, String domainName,
+                                          List<ClaimConfigEntry> requestedClaims) throws UserPortalUIException {
+        List<User> users;
+        if (length < 0) {
+            length = MAX_RECORD_LENGTH;
+        }
+        try {
+            users = getRealmService().getIdentityStore().listUsers(offset, length, domainName);
```

Comment:
if you are handling IllegalStateException in one place need to be handled it in other places as well